src/main/java/reactor/kafka/receiver/internals/DefaultKafkaReceiver.java

```diff
@@ -335,9 +336,11 @@ private void start() {
         subscribeDisposables.add(eventFlux.subscribe(event -> doEvent(event)));
     }
 
-    private void fail(Throwable e) {
-        log.error("Consumer flux exception", e);
-        recordSubmission.error(e);
+    private void fail(Exception e) {
+        ReceiverExceptionHandler.ReceiverExceptionHandlerResponse exceptionHandlerResponse = receiverOptions.receiverExceptionHandler().handle(e);
+        if (ReceiverExceptionHandler.ReceiverExceptionHandlerResponse.FAIL.equals(exceptionHandlerResponse)) {
```

Comment:
Maybe pass recordSubmission to ReceiverExceptionHandler.handle? Then there would be no need to return status